Add pac docs (#1919)
pulumi/docs: content/docs/get-started/policy-as-code/authoring-a-policy-pack.md

                // applies to S3 buckets and asserts that the acl is not "public-read"  nor "public-read-write".
                // If the assertion fails, the custom assertion message will be     displayed to users.
                typedRule(aws.s3.Bucket.isInstance, it => assert.ok(it.acl !==  "public-read"
                    && it.acl !== "public-read-write",
                    "You cannot set public-read or public-read-write on an S3 bucket. "     +
                    "Read more about ACLs here: https://docs.aws.amazon.com/AmazonS3/   latest/dev/acl-overview.html")),
            ],
        }],
    });
    ```

    You can find more example Policy Packs in the [examples repo](https://github.com/pulumi/examples/tree/master/policy-packs). Best practices for writing a Policy Pack can be found [here](../best-practices).

## Testing the Policy Pack Locally

Policy Packs can be tested on a user’s local workstation to facilitate rapid development and testing of policies. This removes the step of publishing and applying policy packs to the Pulumi Console and lets developers reference a policy pack on their local workstation.

1. Run `npm install` in the Policy Pack directory.

> 1. Use the `--policy-pack` flag by going to a directory that contains a Pulumi stack. This flag allows for quicker feedback loops when developing a Policy Pack.

Reviewer: How about something like:

```suggestion
1. Use the `--policy-pack` flag with `pulumi preview` or `pulumi up` to specify the path to the directory containing your Policy Pack when previewing/updating a Pulumi project.
```

>     If you don’t have a stack readily available, you can create a new stack for testing by running `pulumi new aws-typescript` in an empty directory. This AWS example will create an S3 bucket, which is perfect for testing our Policy.

Reviewer: ```suggestion
    If you don’t have a Pulumi project readily available, you can create a new project for testing by running `pulumi new aws-typescript` in an empty directory. This AWS example will create an S3 bucket, which is perfect for testing our Policy.
```

>     In the stack directory run:

Reviewer: ```suggestion
    In the Pulumi project's directory run:
```

    ```sh

>     PULUMI_DEBUG_COMMANDS=true pulumi preview --policy-pack    <path-to-new-policy-pack-directory>

Reviewer: ```suggestion
    $ PULUMI_DEBUG_COMMANDS=true pulumi preview --policy-pack <path-to-policy-pack-directory>
```

    ```

>     If the stack is in compliance, we expect the output to simply tell us which     Policy Packs were run.

Reviewer: ```suggestion
    If the stack is in compliance, we expect the output to simply tell us which Policy Packs were run.
```

>     ```sh

Reviewer: Hugo has a bug with highlighting code fences in lists, which causes this particular block to render incorrectly:

<img width="763" alt="Screen Shot 2019-11-05 at 11 06 06 AM" src="https://user-images.githubusercontent.com/710598/68237647-5fefdb00-ffbc-11e9-9dad-8ed77bf2703d.png">

I generally prefer we keep using the "three backtick` code fences where we can, but I have found a workaround for the cases that don't render correctly.

Using this:

```
    {{< highlight sh >}}
$ PULUMI_DEBUG_COMMANDS=true pulumi preview --policy-pack ~/policy-pack-typescript
Previewing update (dev):

        Type                 Name          Plan
    +   pulumi:pulumi:Stack  test-dev  	create
    +   └─ aws:s3:Bucket     my-bucket     create

Resources:
    + 2 to create

Permalink:
...
{{< /highlight >}}
```

Note:
- The line containing `{{< highlight sh >}}` is indented, which keeps the whole block indented.
- The rest of the lines aren't indented (but they will be indented when rendered).

Will render it correctly as:

<img width="753" alt="Screen Shot 2019-11-05 at 11 08 09 AM" src="https://user-images.githubusercontent.com/710598/68237738-93326a00-ffbc-11e9-8e15-70ddafafd7b1.png">

    $ PULUMI_DEBUG_COMMANDS=true pulumi preview --policy-pack ~/    policy-pack-typescript
    Previewing update (dev):

         Type                 Name          Plan
     +   pulumi:pulumi:Stack  test-dev  	create
     +   └─ aws:s3:Bucket     my-bucket     create

    Resources:
        + 2 to create

    Permalink:
    ...
    ```

1. We can then edit the stack code to specify the ACL to be public-read.

    ```typescript
    const bucket = new aws.s3.Bucket("my-bucket", {
        acl: "public-read",
    });
    ```

> 1. We then run the pulumi preview command again and this time get an error message indicating we failed the preview because of a policy violation.

Reviewer: ```suggestion
1. We then run the `pulumi preview` command again and this time get an error message indicating we failed the preview because of a policy violation.
```

>     ```sh

Reviewer: Same thing here. Due to Hugo bug this renders as:

<img width="768" alt="Screen Shot 2019-11-05 at 11 09 07 AM" src="https://user-images.githubusercontent.com/710598/68237808-c117ae80-ffbc-11e9-8945-6f3026f0e6c8.png">

But specifying as follows will workaround the bug:

```
    {{< highlight sh >}}
$ PULUMI_DEBUG_COMMANDS=true pulumi preview --policy-pack ~/policy-pack-typescript
Previewing update (dev):

        Type                 Name          Plan       Info
    +   pulumi:pulumi:Stack  test-dev  	create     1 error
    +   └─ aws:s3:Bucket     my-bucket     create     1 error

Diagnostics:
    pulumi:pulumi:Stack (test-dev):
    error: preview failed

    aws:s3:Bucket (my-bucket):
    mandatory: [s3-no-public-read] Prohibits setting the publicRead or  publicReadWrite permission on AWS S3 buckets.
    expected value 'true' to == 'false'

Permalink:
...
{{< /highlight >}}
```

    In the Pulumi project's directory run:

    ```sh

>     PULUMI_DEBUG_COMMANDS=true pulumi preview --policy-pack <path-to-policy-pack-directory>

Reviewer: Why remove?

Author: adding back in based on discussions
